change/@ni-nimble-components-a5fd15e6-373b-4e3c-b562-fc2e440df26c.json

```diff
@@ -0,0 +1,7 @@
+{
```

Comment:
I think we are in the right direction. I haven't tried to fit all the things that come to mind together yet so I'm going to layout the puzzle pieces if you want to coalesce them:
- We also want this for textarea: https://github.com/ni/nimble/issues/1228
- And rich text: https://github.com/ni/nimble/pull/1446#discussion_r1310241464
- The feature is coming natively to browsers as a CSS property: https://developer.chrome.com/docs/css-ui/css-field-sizing
- The native feature has semantics about how it relates to other size settings: https://developer.mozilla.org/en-US/docs/Web/CSS/field-sizing#field-sizing_interaction_with_other_size_settings
- I didn't do any research into common attribute names for similar behavior in other design systems
- Want to choose an attribute name that will work well with the native concept, across the other controls, will behave consistently in sizing behavior among all of them, and we can pick up for rich text / text area trivially leveraging the native feature in the future.

Reply:
I don't have answers for everything, but here are some notes from my investigation into some of these topics:

>I didn't do any research into common attribute names for similar behavior in other design systems
I didn't find any common attribute names. These are the attributes I did find:
- [Angular text area](https://material.angular.io/cdk/text-field/overview#automatically-resizing-a-textarea) uses `cdkTextareaAutosize` "attribute"
- [Shoelace text area](https://shoelace.style/components/textarea#expand-with-content) uses `resize` attribute with possible values of `"none"`, `"auto"`, or `"vertical"`
- [PrimeNG](https://primeng.org/inputtextarea#autoresize) has a boolean attribute `autoResize` attribute
- The text area didn't appear to be resizable in the Carbon design system

>Curious if we have a story of how it may interact with CSS resize which certainly makes sense for textarea and richtext and I think actually could make sense for a data table as well
- I tried out CSS resize on the table with & without auto sizing, and it worked well in both cases. Specifically, in both cases, the table correctly adjusted the visible area of rows based on the height of the table. I temporarily added a control to the table storybook page if you wanted to play around with it. So, I think the "story" of how the two features interact is that the table will auto size by default, but interactively changing the size of the table would take precedence over the automatically determined size. I think something similar would also make sense for other components.

Reply:
@rajsite, given that I couldn't find any standard way for components to be marked as auto resizing, do you have any thoughts on how we should move forward? Any objections to using a boolean attribute of `autoHeight` as in this PR?

Reply:
The thing I'm on the fence about is how the api could grow and be consistent across controls:

Consistent apis:
- table today: auto-height
- table and other controls (textarea, etc) tomorrow: auto-height, auto-width
  - auto-width on table would size so all columns show
  - auto-height and auto-width would have to be implemented on top of `field-sizing: content` which on it's own grows height and width.

Inconsistent apis:
- table today: auto-height
- table tomorrow: auto-height, auto-width
- Other controls: fit-to-content? auto-resize?
  - Rely on user setting bounds on height and width, just fill the space

Another thing that just "feels" wrong is having styling, aka sizing, be part of the attribute api. I've fought pretty hard against that (i.e. attributes for "small" / "medium"/ "large" for spinners or dialog). I think ideally I'd like to say textarea / rich-text are just text areas. So in your style for sizing set `field-sizing: content` like you would any other text field. If you swap a native text-field for ours you can size it with CSS, you don't need to impact sizing and layout with a combination of attributes and CSS. (talked offline to express this concept more)

Reply:
Played around with the latest changes in storybook and it feels excellent. Implementation looks good, only nit would be a preference for the simpler name `--ni-nimble-table-fit-rows-height` which is a nice analog to the [`fit-content`](https://developer.mozilla.org/en-US/docs/Web/CSS/fit-content) property.

Docs pattern can be similar to how we document dialog sizing style. And maybe a story test (separate from the matrix) with a couple sizes of data to show the property is behaving correctly (something like 5, 10, 45 or something)